src/abstractions/log.service.ts

```diff
@@ -1,11 +1,49 @@
 import { LogLevelType } from '../enums/logLevelType';
+import { ConsoleLogService } from '../services/consoleLog.service';
 
 export abstract class LogService {
-    debug: (message: string) => void;
-    info: (message: string) => void;
-    warning: (message: string) => void;
-    error: (message: string) => void;
-    write: (level: LogLevelType, message: string) => void;
+
+    private static get instance() {
+        if (this._instance == null) {
+            this._instance = new ConsoleLogService(false);
```

Comment:
I know this provides a really nice fallback, and I get that, however I'm struggling with having a Singleton pattern implementation of this abstract type which has a circular reference to a descendent type/default implementation of itself. Kind of outside of established OOD principals of encapsulation and inheritance (although a great example of polymorphism 😁 ). 

It seems sucky, I know, but I would have more warm & fuzzies if we simply left `this._instance` as `null` if not explicitly set anywhere by a downstream runtime/client, and following the "quasi" DI pattern, each client app is responsible for and **must** set this in order for logging to be "a thing"; otherwise calls to `.debug`, `.error`, etc. are simply ignored within the static methods if no logging instance is set. I would simply like to avoid circular references in the dependency chain.